### PR TITLE
[prometheus-pushgateway] Support dual stack clusters

### LINKED ISSUE #4423: [prometheus-pushgateway] Support dual stack clusters
### Is your feature request related to a problem ?

There is no way for me to set the ipFamilyPolicy to "preferDualStack" in the current helm chart

### Describe the solution you'd like.

Option to define `ipFamilyPolicy` in helm chart

### Describe alternatives you've considered.

Patching svc with kustomize

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by M0NsTeRRR**
#### What this PR does / why we need it
add support for dual stack clusters

#### Which issue this PR fixes

- fixes #4423

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @M0NsTeRRR, for your PR. Will you please reuse the code already present in some charts? E.g. prometheus-node-exporter - [values](https://github.com/prometheus-community/helm-charts/blob/d26d47b5721ad6cadc3835a1275f60e13b0bd0e4/charts/prometheus-node-exporter/values.yaml#L97) and service [template](https://github.com/prometheus-community/helm-charts/blob/d26d47b5721ad6cadc3835a1275f60e13b0bd0e4/charts/prometheus-node-exporter/templates/service.yaml#L14). That would be appreciated, indeed.

**Comment by M0NsTeRRR** [comment]
Hello,
Sure will rework it to match existing works, thanks for your review :)

File: charts/prometheus-pushgateway/templates/service.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ spec:
   {{ else if and .Values.runAsStatefulSet (not (has .Values.service.type $stsNoHeadlessSvcTypes)) }}
   clusterIP: None # Headless service
   {{- end }}
+  {{- if .Values.service.ipDualStack.enabled }}
+    ipFamilies: {{ toYaml .Values.service.ipDualStack.ipFamilies | nindent 4 }}
+    ipFamilyPolicy: {{ .Values.service.ipDualStack.ipFamilyPolicy }}
```
Comment (zeritti) [inline charts/prometheus-pushgateway/templates/service.yaml:24]: Indentation goes too far:

```suggestion
  ipFamilies: {{ toYaml .Values.service.ipDualStack.ipFamilies | nindent 4 }}
  ipFamilyPolicy: {{ .Values.service.ipDualStack.ipFamilyPolicy }}
```

**Review by zeritti — CHANGES_REQUESTED** [review]

File: charts/prometheus-pushgateway/templates/service.yaml (hunk 1 above)
Comment (M0NsTeRRR) [reply charts/prometheus-pushgateway/templates/service.yaml]: Didn't see it on my IDE, good catch 👍

**Review by zeritti — APPROVED** [review]
Thank you for your changes, LGTM.

**Pull request merged**